src/components/user/UpdateUserProfileFields.js

```diff
@@ -0,0 +1,201 @@
+        });
+        // Handle success (e.g., show a success message)
+        console.log('User profile updated:', response.data);
+        setErrors({}); // Clear any previous errors
+        setSuccessMessage('User profile saved successfully.'); // Set success message
+        setFormData({ // Clear the form
+          name: '',
+          email: '',
+          oldPassword: '',
+          newPassword: '',
+        });
+      } catch (error) {
+        // Handle errors and update the state with validation errors
+        setErrors(error.response.data.errors);
+        setSuccessMessage(''); // Clear any previous success message
+      }
+    }
+  };
+
+  const handlePasswordSubmit = async (e) => {
+    e.preventDefault();
+    if (!formData.oldPassword || !formData.newPassword) {
+      // Check if either oldPassword or newPassword is empty
+      setErrors({
+        oldPassword: !formData.oldPassword ? "Old Password is required" : "",
+        newPassword: !formData.newPassword ? "New Password is required" : "",
+      });
+    } else {
+      try {
+        // Call the backend API to update the user's password
+        const response = await axiosInstance.patch('/users/updateUserPassword', {
+          oldPassword: formData.oldPassword,
+          newPassword: formData.newPassword,
+        });
+        // Handle success (e.g., show a success message)
+        console.log('Password updated:', response.data);
+        setSuccessMessagePassword('Password changed successfully.'); // Set success message for password change
+        setErrors({}); // Clear any previous errors
+        setSuccessMessage(''); // Clear any previous success message for profile update
+        setFormData({ // Clear the form
+          name: '',
+          email: '',
+          oldPassword: '',
+          newPassword: '',
+        });
+      } catch (error) {
+        // Handle errors and update the state with validation errors
+        setErrors(error.response.data.errors);
+        setSuccessMessagePassword(''); // Clear any previous success message for password change
+      }
+    }
+  };
+
+  return (
+    <Card className="mt-2 border-0 rounded-0 shadow-sm">
+      <CardContent>
+        <h3 className="text-uppercase">My Profile</h3>
+        <div className="text-center">
+          <Avatar
+            src={require('../../images/customer.png')}
+            alt="user profile"
+            className="img-fluid rounded-circle"
+            sx={{
+              width: '100px',
+              height: '100px',
+              maxWidth: '100px',
+              maxHeight: '100px',
+            }}
+          />
+        </div>
+        {successMessage && (
+          <div className="text-success">{successMessage}</div>
+        )}
+        {successMessagePassword && (
+          <div className="text-success" >{successMessagePassword}</div>
+        )}
+        <form onSubmit={handleProfileSubmit}>
+          <Table responsive striped hover className="text-center mt-5">
```

Comment:
There's errors in the console due to these fields. I think when the `UserProfile.js` component was made that used these fields, we were planning to go back to fix that.

The errors are due to these fields not being valid for material-ui (I think when the UserProfile component was used there was some confusion between the syntax for styling material-ui table vs bootstrap-based table).

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/99532ee7-e8fc-46ef-bc76-ebf57170c495)

```suggestion
          <Table className="text-center mt-5">
```

Same for the `<Table>` component on line 164